### ISSUE TITLE
I18n::InvalidLocale: :en is not a valid locale

### ISSUE THREAD

**Issue by mrbongiolo** [dry-rb/dry-validation#216]
This happened when upgrading from `0.8.0` to `0.9.x`.

In my rails app I have the I18n locales defined like that, on application.rb file:
```ruby
    I18n.config.enforce_available_locales = true
    config.i18n.available_locales = ["pt-BR"]
    config.i18n.default_locale = "pt-BR"
```

The current version of dry-v ignores the `I18n.default_locale` option in favor of it's `:locale` option (that can be set when calling `results.messages`), since I haven't used the `:locale` option there then it defaulted to `:en`.

Current solution: call `results.messages(locale: 'pt-BR')`

Suggestion: add a config option to set the "default" locale, similar to the way we can set `messages = :i18n`.

**Comment by agustin** [comment]
Hi! I'm also struggling with a similar situation in which `I18n` is configured and without monkey patching **[Reform](https://github.com/apotonick/reform)** ([Issue #391](https://github.com/apotonick/reform/issues/391)) I'm not able to pass the `locale:` option to the `messages(...)` but working on it I encounter that the there is something unclear on why is implemented the way it is, and I believe that I'm missing something or have a lack of background, that's why I'm asking in here.

I wonder the following, why someone will define the use of `:i18n` for the messages:
```ruby
Dry::Validation.Schema do
  configure do
    config.messages = :i18n
  end
  #[...]
end
```
and then need to pass it as an argument to the `messages(...)`, shouldn't make more sense in this case to follow the configurations on the `I18n` Module? like the `I18n.locale` or `I18n.default_locale`?
I worked on a commit for this approach, but thought that maybe there is a reason that I'm not aware of that justify this behavior. (Happy to push the commit and discuss it over there also)

Anyone knows? maybe @solnic can give us some better insight on this 🔦 😃 
Thanks!

**Comment by solnic** [comment]
This is a simple fix probably. There's a mistake in the logic as message compiler defaults to `:en` [right here](https://github.com/dry-rb/dry-validation/blob/master/lib/dry/validation/message_compiler.rb#L14) so the only thing we need to do is to add `Messages::Abstract#default_locale` and implement it in `Messages::YAML` and `Messages::I18n` and tweak that line to do `@locale = options.fetch(:locale, messages.default_locale)`.

**Comment by cavi21** [comment]
That sounds perfect @solnic! do you want me to create a PR for that? or you rather do it by yourself?
Don't hesitate to let me know, happy to help.

**Comment by solnic** [comment]
@cavi21 a PR would be great :)

**Comment by cavi21** [comment]
I believe that this issue is solved on the [version **v0.9.5**](https://github.com/dry-rb/dry-validation/blob/master/CHANGELOG.md#v095-2016-08-16) 

Maybe this could be close

**Comment by solnic** [comment]
Yep it's fixed in 0.9.5
